src/linalg.jl

```diff
@@ -365,6 +365,68 @@ const _length_limit = Length(200)
     end
 end
 
+@inline kron(a::StaticVector, b::StaticVector) = _kron_vec_x_vec(_length_limit, Size(a), Size(b), a, b)
+@generated function _kron_vec_x_vec(::Length{length_limit}, ::Size{SA}, ::Size{SB}, a, b) where {length_limit,SA,SB}
+    outsize = SA .* SB
+    if prod(outsize) > length_limit
+        return :( SizedVector{$(outsize[1])}( kron(drop_sdims(a), drop_sdims(b)) ) )
+    end
+    rows = [:(vcat($([:(a[$(LinearIndices(SA)[i])]*b) for i=1:SA[1]]...)))]
+    return quote
+        @_inline_meta
+        @inbounds return $(rows...)
+    end
+end
+
+@inline kron(a::RowVector{<:Number,<:StaticArray}, b::RowVector{<:Number,<:StaticArray}) = _kron_vec_x_vec(_length_limit, Size(a'), Size(b'), a', b')'
+
+@inline kron(a::RowVector{<:Number,<:StaticArray}, b::StaticVector) = _kron_tvec_x_vec(_length_limit, Size(a), Size(b), a, b)
+@generated function _kron_tvec_x_vec(::Length{length_limit}, ::Size{SA}, ::Size{SB}, a, b) where {length_limit,SA,SB}
+    outsize = (SA[1] * SB[1], SA[2])
+    if prod(outsize) > length_limit
+        return :( SizedMatrix{$(outsize[1]),$(outsize[2])}( kron(drop_sdims(a), drop_sdims(b)) ) )
+    end
+    rows =  [:(hcat($([:(a[$(LinearIndices(SA)[i])]*b) for i=1:SA[2]]...)))]
+    return quote
+        @_inline_meta
+        @inbounds return $(rows...)
+    end
+end
+
+@inline kron(a::StaticVector, b::RowVector{<:Number,<:StaticArray}) = _kron_tvec_x_vec(_length_limit, Size(a'), Size(b'), a', b')'
+
+
+@inline kron(a::StaticVector, b::StaticMatrix) = _kron_vec_x_mat(_length_limit, Size(a), Size(b), a, b)
+@generated function _kron_vec_x_mat(::Length{length_limit}, ::Size{SA}, ::Size{SB}, a, b) where {length_limit,SA,SB}
+    outsize = (SA[1] * SB[1], SB[2])
+    if prod(outsize) > length_limit
+        return :( SizedMatrix{$(outsize[1]),$(outsize[2])}( kron(drop_sdims(a), drop_sdims(b)) ) )
+    end
+    rows = [:(vcat($([:(a[$(LinearIndices(SA)[i])]*b) for i=1:SA[1]]...)))]
+    return quote
+        @_inline_meta
+        @inbounds return $(rows...)
+    end
+end
+
+@inline kron(a::RowVector{<:Number,<:StaticArray}, b::StaticMatrix) = _kron_vec_x_mat(_length_limit, Size(a'), Size(b'), a', b')'
+
+@inline kron(a::StaticMatrix, b::StaticVector) = _kron_mat_x_vec(_length_limit, Size(a), Size(b), a, b)
+@generated function _kron_mat_x_vec(::Length{length_limit}, ::Size{SA}, ::Size{SB}, a, b) where {length_limit,SA,SB}
+    outsize = (SA[1] * SB[1], SA[2])
+    if prod(outsize) > length_limit
+        return :( SizedMatrix{$(outsize[1]),$(outsize[2])}( kron(drop_sdims(a), drop_sdims(b)) ) )
+    end
+    #rows = [:(vcat($([:(a[$(LinearIndices(SA)[i])]*b) for i=1:SA[1]]...)))]
+    rows = [:(hcat($([:(a[$(LinearIndices(SA)[i, j])]*b) for j=1:SA[2]]...))) for i=1:SA[1]]
+    return quote
+        @_inline_meta
+        @inbounds return vcat($(rows...))
+    end
+end
+
+@inline kron(a::StaticMatrix, b::RowVector{<:Number,<:StaticArray}) = _kron_mat_x_vec(_length_limit, Size(a'), Size(b'), a', b')'
```

Comment:
Probably more direct to reshape `b` into a matrix... but I don't think we have a method for that?

Reply:
Are you suggesting that we could reshape `b` into a `1 x 3` matrix and then dispatch on the `kron` that operates on two static matrices? 

It seems that there is no method currently defined for that. 
For example, with 
```julia
s = @SVector [1, 2,  3];
reshape(s',(1,3))

1×3 Base.ReshapedArray{Int64,2,RowVector{Int64,StaticArrays.SArray{Tuple{3},Int64,1,3}},Tuple{}}:
 1  2  3
```
I get this `Base.ReshapedArray` type instead of the required `StaticMatrix`.